From 54243c9557bc2f567d7f21b657e053710ee3d4f9 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:13:02 -0400
Subject: [powerpc] pseries: hcall functions are exported to modules, need _GLOBAL_TOC()

Message-id: <1410545655-205645-554-git-send-email-dzickus@redhat.com>
Patchwork-id: 94438
O-Subject: [RHEL7 PATCH 553/626] powerpc/pseries: hcall functions are exported to modules, need _GLOBAL_TOC()
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit c1931e21816a031a5aff7084e7f10eb84e24d8cc
Author: Anton Blanchard <anton@samba.org>
Date:   Tue May 13 20:48:57 2014 +1000

    powerpc/pseries: hcall functions are exported to modules, need _GLOBAL_TOC()

    The hcall macros may call out to c code for tracing, so we need
    to set up a valid r2. This fixes an oops found when testing
    ibmvscsi as a module.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/hvCall.S b/arch/powerpc/platforms/pseries/hvCall.S
index 7891a86..99ecf0a 100644
--- a/arch/powerpc/platforms/pseries/hvCall.S
+++ b/arch/powerpc/platforms/pseries/hvCall.S
@@ -106,7 +106,7 @@ END_FTR_SECTION(0, 1);      \
 
  .text
 
-_GLOBAL(plpar_hcall_norets)
+_GLOBAL_TOC(plpar_hcall_norets)
  HMT_MEDIUM
 
  mfcr r0
@@ -122,7 +122,7 @@ _GLOBAL(plpar_hcall_norets)
  mtcrf 0xff,r0
  blr    /* return r3 = status */
 
-_GLOBAL(plpar_hcall)
+_GLOBAL_TOC(plpar_hcall)
  HMT_MEDIUM
 
  mfcr r0
@@ -188,7 +188,7 @@ _GLOBAL(plpar_hcall_raw)
 
  blr    /* return r3 = status */
 
-_GLOBAL(plpar_hcall9)
+_GLOBAL_TOC(plpar_hcall9)
  HMT_MEDIUM
 
  mfcr r0
-- 
1.7.1